Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -23,9 +23,18 @@ public MemberPath(IMember member, string parentPath)
         }
 
         public MemberPath(Type reflectedType, Type declaringType, string dottedPath)
+            : this(dottedPath)
         {
             this.reflectedType = reflectedType;
             this.declaringType = declaringType;
+        }
+
+        public MemberPath(string dottedPath)
```

Comment:
I haven't dug into whether it's a realistic scenario, but if `IsSameAs(MemberPath candidate)` is called on an instance created from this constructor we could throw an NRE as `declaredType` might be null.

So as an extra precaution
```cs
declaringType?.IsAssignableFrom(candidate.reflectedType) == true
```

Reply:
It's not a valid scenario, but I do agree the logic in `MemberPath` has become a bit fishy. I couldn't come up with a nicer solution yet.